server/src/main/java/io/crate/metadata/doc/DocTableInfo.java

```diff
@@ -485,4 +504,31 @@ public String toString() {
     public Collection<ColumnIdent> notNullColumns() {
         return notNullColumns;
     }
+
+    /**
+     * Starting from 5.5 column OID-s are used as source keys.
+     * Even of 5.5, there are no OIDs (and thus no source key rewrite happening) for:
+     * <ul>
+     *  <li>OBJECT (IGNORED) sub-columns</li>
+     *  <li>Empty arrays, or arrays with only null values</li>
+     *  <li>Internal object keys of the geo shape column, such as "coordinates", "type"</li>
+     * </ul>
+     */
+    public Function<String, String> lookupNameBySourceKey() {
```

Comment:
Isn't this `byStorageIdent` ? Or why the different naming here?

Reply:
storage ident can be prefixed, so what we have in source key not always equal to storage ident

Reply:
Actually, I'd even favour if we could get the prefix into the `storageIdent` function, but as the prefix depends on the parent column or table version sometimes, it's not easily possible. I'd already note down to follow up here.